llvm/lib/Target/AArch64/AArch64Arm64ECCallLowering.cpp

```diff
@@ -414,8 +416,7 @@ Function *AArch64Arm64ECCallLowering::buildExitThunk(FunctionType *FT,
     // with an attribute.)
     //
     // The first argument is the called function, stored in x9.
-    if (Arg.getType()->isArrayTy() || Arg.getType()->isStructTy() ||
-        DL.getTypeStoreSize(Arg.getType()) > 8) {
+    if (Arg.getType() != X64ArgType) {
```

Comment:
If the translation here is going to get more complicated, maybe we should consider making getThunkType explicitly return whether the argument is supposed to be passed indirectly, rather than trying to infer it like this.

Reply:
I went ahead and did this: my nicer to have all of the logic centralized in a single function.